src/DependencyManagement/DependencyManagementUtils.cs

```diff
@@ -0,0 +1,117 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Net;
+using System.Xml;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.DependencyManagement
+{
+    internal class DependencyManagementUtils
+    {
+        // PSGallery url to query for the latest module version.
+        private const string PsGalleryUri = "https://www.powershellgallery.com/api/v2/FindPackagesById()?id='moduleName'";
+
+        /// <summary>
+        /// Returns true if the given majorVersion is less or equal to the major version in latestSupportedVersion.
+        /// </summary>
+        internal static bool IsValidMajorVersion(string majorVersion, string latestSupportedVersion)
+        {
+            // A Version object cannot be created with a single digit so add a '.0' to it.
+            var requestedVersion = new Version(majorVersion + ".0");
+            var latestVersion = new Version(latestSupportedVersion);
+
+            var result = (requestedVersion.Major <= latestVersion.Major);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Deletes the contents at the given directory.
+        /// </summary>
+        internal static void EmptyDirectory(string path)
+        {
+            var directoryInfo = new DirectoryInfo(path);
+
+            if (directoryInfo.Exists)
+            {
+                foreach (var file in directoryInfo.GetFiles())

```

Comment:
Is it possible to remove the directory recursively if exist, and create again. In this case, we don't need to do a forloop.
```
if (directoryInfo.Exists) {
    directoryInfo.Delete(true);
}

directoryInfo.Create();
```

Reply:
@Hazhzeng : Good point. However, there is a known issue sporadically when the directory is not empty. This is documented here: https://stackoverflow.com/questions/329355/cannot-delete-directory-with-directory-deletepath-true

Reply:
@Francisco-Gamino Thanks for clarification 👍

Reply:
@Hazhzeng : Thank you for taking a look!